### PR TITLE
ref(server): introduce client name enum

### PR DESCRIPTION
**Pull request by Litarnus**
Introduces an enum which allows us to have more fine grained control over which sdks are tagged

#skip-changelog

### CODE REVIEW

File: relay-server/src/statsd.rs [hunk 1]
```diff
@@ -973,3 +973,83 @@ impl<S: AsRef<str>> From<S> for PlatformTag {
         }
     }
 }
+
+pub enum SdkTag {
```
Comment (jjbayer) [inline relay-server/src/statsd.rs]:
```suggestion
/// Low-cardinality SDK name that can be used as a statsd tag.
pub enum SdkTag {
```

**Review by jjbayer — APPROVED** [review]
Nice!

File: relay-server/src/endpoints/common.rs [hunk 2]
```diff
@@ -334,16 +334,15 @@ pub async fn handle_envelope(
     state: &ServiceState,
     envelope: Box<Envelope>,
 ) -> Result<Option<EventId>, BadStoreRequest> {
-    let client_name = envelope
+    let client_name: SdkTag = envelope
         .meta()
         .client_name()
```
Comment (Dav1dde) [inline relay-server/src/endpoints/common.rs]: `client_name` could already return the `SdkTag` (probably would name it just `Sdk` or `ClientName`)

File: relay-server/src/statsd.rs [hunk 3]
```diff
@@ -973,3 +973,83 @@ impl<S: AsRef<str>> From<S> for PlatformTag {
...
+    NextJs,
+    Node,
+    React,
+    Vue,
+    Native,
+    Laravel,
+    Symfony,
+    Php,
+    Python,
+    Other,
```
Comment (Dav1dde) [inline relay-server/src/statsd.rs]: If you change `Other` to `Other(String)` or `Other(&'a str)` we can still recover the original full name. We use something similar for metric units and envelope items already.

**Review by Dav1dde — APPROVED** [review]

**Review by Dav1dde — APPROVED** [review]
Nice! :shipit:

File: relay-server/src/endpoints/common.rs (hunk 2 above)
Comment (jjbayer) [reply relay-server/src/endpoints/common.rs]: IMO mapping a high-cardinality client name to a low cardinality name is purely a concern of the statsd module and a technicality of emitting metrics, it is not an inherent property of `RequestMeta`.

**Pull request merged**